MetaMask/metamask-extension
# UX Snaps: Updated dapp connection state for connected snaps

This PR is to ensure that when the dapp is connected with a snap, the site connection icon should indicate a connection. Similarly, the currently connected site modal doesn't indicate that the site is connected.

* Fixes #20792 

## Screenshots/Screencaps

### After

## Both Account and Snap are connected




https://github.com/MetaMask/metamask-extension/assets/39872794/106e3870-2d6b-4ef7-8d68-51d93d704a42




### Only Snap Connected




https://github.com/MetaMask/metamask-extension/assets/39872794/aa4acfcc-9d08-4b5d-986f-9d28469123e5





## Manual Testing Steps

- Go to https://snaps.metamask.io/
- Connect a snap
- Open MetaMask in popup view, check if the connectivity circle is hollow green
- Click on the circle, the snap should be in the popover menu.
- For only snaps connected state, permissions should not be there in modal
- Go to https://app.walletchat.fun/
- Sign in using MetaMask
- Open MetaMask in popup view, check if the connectivity circle should be filled green
- Click on the circle, both the account and snap should be in the popover menu

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
